src/runtime/auto-prompt-manager.ts

```diff
@@ -270,76 +282,71 @@ export class AutoPromptManager {
   }
 
   /**
-   * Determines whether a mini prompt for contributions or subscriptions should
-   * be shown.
+   * Determines whether a mini prompt for contributions or subscriptions can
+   * be shown based on the UI Predicates.
    */
-  async shouldShowAutoPrompt_(
-    clientConfig: ClientConfig,
-    entitlements: Entitlements,
-    autoPromptType?: AutoPromptType
-  ): Promise<boolean> {
+  canDisplayMonetizationPromptFromUiPredicates_(
+    uiPredicates?: UiPredicates
+  ): boolean {
     // If false publication predicate was returned in the response, don't show
     // the prompt.
-    if (
-      clientConfig.uiPredicates &&
-      !clientConfig.uiPredicates.canDisplayAutoPrompt
-    ) {
-      return Promise.resolve(false);
+    if (uiPredicates && !uiPredicates.canDisplayAutoPrompt) {
+      return false;
     }
+    return true;
+  }
 
+  /**
+   * Determines whether a monetization prompt should be shown as a paywall,
+   * meaning with the explicit intent to hard- or soft-restrict access to the
+   * page. Does not prevent a monetization prompt from displaying as an
+   * eligible audience action.
+   */
+  async shouldShowMonetizationPromptAsPaywall(
```

Comment:
Hi @ChrisAntaki , wanted to get your take on the naming of this function and boolean `monetizationPromptWasDisplayedAsPaywall_`.

TLDR: The term `Paywall` is used to convey the intent to restrict a reader's access to some content [old]. This is different from swg.js displaying a monetization prompt as an audience action provided from the article endpoint [newly added].

Previously, we computed whether to show a monetization prompt ONLY from a few factors: client config, entitlements, and any frequency caps (explicitly distinct from the audience actions). With our upcoming June launch, we are expanding monetization prompts to also be served as audience actions. As such, we want to now distinguish these two serving pathways; the idea being that the monetization prompts can be shown with the intent to paywall content either as a hard-restriction (non-dismissible prompt) or soft-restriction (dismissible prompt), vs. when a monetization prompt is the next eligible audience action. Note: all Paywalls are Monetization Prompts, but not all Monetization Prompts are Paywalls.

Thoughts on this terminology?

Reply:
Thanks for the great summary. I like the new name and distinction from audience action serving path in the comment.

Would it be more accurate to narrow down the name to "soft paywall"?
One of the condition is returning `false` when the page is locked, meaning it doesn't cover showing "hard-" paywall.

Reply:
I'm liking this clear naming